lib/visitIdentifierNodes.js

```diff
@@ -102,10 +116,9 @@ export default function visitIdentifierNodes(
       queue.unshift(...itemsToAdd);
       continue; // eslint-disable-line no-continue
     }
-    const normalizedNode = normalizeNode(current.node, current.context);
-    if (normalizedNode) {
+    normalizeNode(current.node, current.context, (normalizedNode) => {
       visitor(normalizedNode, current.context);
```

Comment:
> optionally returning an array
Having mixed return values have a bit of a code smell for me, having to test the object in order to use it. This way the function is more honest: "I'm going to produce zero, one or several normalizedNodes".

Of course, we could always return an array. Not sure why I went with the callback pattern. I don't like it, I never liked it :) An iterable of some kind makes the most sense. I'll convert it to a generator if you agree this would be a most pleasant way forward?

Reply:
I don't have a strong preference, you make the call :)